jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -1148,6 +1150,18 @@ protected void doStop() throws Exception
         getServletContext().removeAttribute(DecoratedObjectFactory.ATTR);
     }
 
+    protected ServletContextRequest newServletContextRequest(ServletChannel servletChannel,
+                                                             Request request,
+                                                             Response response,
+                                                             String pathInContext,
+                                                             ServletHandler.MappedServlet mappedServlet,
+                                                             PathSpec pathSpec,
+                                                             MatchedPath matchedPath)
+    {
+        return new ServletContextRequest(_servletContext, servletChannel, request, response, pathInContext,
+            mappedServlet, pathSpec, matchedPath);
+    }
```

Comment:
Maybe this is less fat:
```suggestion
    protected ServletContextRequest newServletContextRequest(ServletChannel servletChannel,
                                                             Request request,
                                                             Response response,
                                                             String pathInContext,
                                                             MatchedResource<ServletHandler.MappedServlet> matchedServlet)
    {
        return new ServletContextRequest(_servletContext, servletChannel, request, response, pathInContext,
            matchedServlet.getResource(), matchedServlet.getPathSpec(), matchedServlet.getMatchedPath());
    }
```

maybe even pass into constructor like that?

Reply:
`MatchedResource<ServletHandler.MappedServlet>` is not retained and usable for folks that are wrapping this class.

Reply:
I don't understand? `MatchedResource<ServletHandler.MappedServlet>` is a public class and contains a: ServletHandler.MappedServlet, PathSpec & MatchedPath, so why pass them separately when they can be passed simply in a single public class?

I think it definitely can be done and folks that wrap the class will get the info they need one way or another.   The only question is do we want to do it or not?   PRO: not so fat signature, CON: moderately verbose generic class.  I'm 60:40 for the thinner API.